packages/commons/core-utils/src/mergeWithOverrides.ts

```diff
@@ -12,32 +25,68 @@ export function mergeWithOverrides<T extends object>({ data, overrides }: { data
             : undefined
     ) as T;
     // Remove any nullified values
-    const filtered = omitDeepBy(merged, isNull) as T;
-    return filtered;
+    const filtered = omitDeepBy(merged, isNull, {
+        ancestorKeys: nullOmissionKeys ?? [],
+        isDescendant: false
+    });
+    return filtered as T;
 }
 
 // This is essentially lodash's omitBy, but actually running through your object tree.
 // The logic has been adapted from https://github.com/siberiacancode/lodash-omitdeep/tree/main.
 interface OmitDeepBy {
-    <T>(object: Dictionary<T> | null | undefined, predicate?: ValueKeyIteratee<T>): Dictionary<T>;
-    <T>(object: NumericDictionary<T> | null | undefined, predicate?: ValueKeyIteratee<T>): NumericDictionary<T>;
-    <T extends object>(object: T | null | undefined, predicate: ValueKeyIteratee<T[keyof T]>): PartialObject<T>;
+    <T>(
+        object: Dictionary<T> | null | undefined,
+        predicate?: ValueKeyIteratee<T>,
+        ancestorOmissionCriteria?: AncestorOmissionCriteria
+    ): Dictionary<T>;
+    <T>(
+        object: NumericDictionary<T> | null | undefined,
+        predicate?: ValueKeyIteratee<T>,
+        ancestorOmissionCriteria?: AncestorOmissionCriteria
+    ): NumericDictionary<T>;
+    <T extends object>(
+        object: T | null | undefined,
+        predicate: ValueKeyIteratee<T[keyof T]>,
+        ancestorOmissionCriteria?: AncestorOmissionCriteria
+    ): PartialObject<T>;
 }
 
-// eslint-disable-next-line  @typescript-eslint/no-explicit-any
-export const omitDeepBy: OmitDeepBy = (object: unknown, cb: any): any => {
+export const omitDeepBy: OmitDeepBy = (
+    object: unknown,
+    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+    cb: any,
+    ancestorOmissionCriteria?: AncestorOmissionCriteria
+    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+): any => {
     function omitByDeepByOnOwnProps(object: unknown) {
         if (Array.isArray(object)) {
-            return object.map((element) => omitDeepBy(element, cb));
+            return object.map((element) => omitDeepBy(element, cb, ancestorOmissionCriteria));
         }
 
         if (isPlainObject(object)) {
             const temp: Record<string, unknown> = {};
             // eslint-disable-next-line  @typescript-eslint/no-explicit-any
             for (const [key, value] of Object.entries<Record<string, PropertyName | object>>(object as any)) {
-                temp[key] = omitDeepBy(value, cb);
+                temp[key] = omitDeepBy(
+                    value,
+                    cb,
+                    ancestorOmissionCriteria != null
+                        ? {
+                              ...ancestorOmissionCriteria,
+                              isDescendant:
```

Comment:
Discussed offline - we might be able to drop the ancestorKeys breadcrumbs in favor of detecting if _this_ key is part of the "allow keys", then pass it in from there.

Defer to you on whether or not this refactor makes sense.